src/parser.rs

```
@@ -38,15 +39,21 @@ impl<'a> Packet<'a> {
             nameservers.push(try!(parse_record(data, &mut offset)));
         }
         let mut additional = Vec::with_capacity(header.additional as usize);
+        let mut opt = None;
         for _ in 0..header.additional {
-            additional.push(try!(parse_record(data, &mut offset)));
+            if data[offset..offset+3] == OPT_RR_START {
+                opt = parse_opt_record(data, &mut offset).ok();
```

Why do you ignore the error?